File: frontend_rebuild/types/media.ts
export type MediaResourceType =
  | "image"
  | "video";


export type MediaScope =
  | "listings"
  | "agent-documents"
  | "profiles";


export type MediaDeliveryType =
  | "upload"
  | "authenticated";


export interface MediaUploadSignature {
  cloud_name: string;

  api_key: string;

  timestamp: number;

  signature: string;

  folder: string;

  resource_type:
    MediaResourceType;

  delivery_type:
    MediaDeliveryType;

  allowed_formats: string;

  upload_url: string;
}


export interface CloudinaryUploadResponse {
  secure_url: string;

  public_id: string;

  resource_type:
    MediaResourceType;

  /**
   * Actual Cloudinary format,
   * e.g. jpg, png, webp,
   * mp4.
   */
  format: string;

  /**
   * Cloudinary delivery type.
   *
   * Normal media:
   *   upload
   *
   * Agent verification:
   *   authenticated
   */
  type: MediaDeliveryType;

  bytes: number;

  width?: number;

  height?: number;

  duration?: number;
}


/**
 * Returned after the browser has
 * completed the Cloudinary upload.
 */
export interface UploadedMedia {
  url: string;

  publicId: string;

  resourceType:
    MediaResourceType;

  fileFormat: string;

  deliveryType:
    MediaDeliveryType;

  originalName: string;

  bytes: number;
}


/**
 * Media state used while a listing
 * form is being edited.
 */
export interface MediaValue {
  url: string;

  publicId:
    string | null;
}

File: backend_rebuild/app/schemas/media.py
from typing import Literal

from pydantic import BaseModel


MediaResourceType = Literal[
    "image",
    "video",
]

MediaScope = Literal[
    "listings",
    "agent-documents",
    "profiles",
]

MediaDeliveryType = Literal[
    "upload",
    "authenticated",
]


class MediaUploadSignatureRequest(BaseModel):
    resource_type: MediaResourceType
    scope: MediaScope


class MediaUploadSignatureResponse(BaseModel):
    cloud_name: str
    api_key: str
    timestamp: int
    signature: str
    folder: str

    resource_type: MediaResourceType
    delivery_type: MediaDeliveryType

    allowed_formats: str
    upload_url: str